Changed Port Number and Fixed Bug in Create Class Forms

midhunann opened the pull request:
## Changes & Fixes

### 1. Port Change  
- Updated server port from **5000 → 8000**.

### 2. Bug Fix: Create Class Form  
####  **Fixes & Enhancements**  
- **Added `subjectDropdownRef`** to track the dropdown container.  
- **Implemented Click-Outside Detection** using `useEffect`:  
  - Listens for `mousedown` events when the dropdown is open.  
  - Closes dropdown when clicking outside.  
  - Cleans up event listener on unmount.  

#### **UI Improvements**  
- **Dropdown Enhancements**:  
  - Added a dropdown arrow indicator for clarity.  
  - Introduced a **"Done"** button for explicit closure.  
  - Improved dropdown header styling.  
- **Multi-Selection Support**:  
  - Dropdown remains open while toggling subjects.

Aashiq-Edavalapati commented on frontend/src/app/teacher/create-class/page.js:
```diff
@@ -253,22 +274,35 @@ export default function CreateClass() {
             
             {/* Subjects */}
-            <div className="mb-6 relative">
+            <div className="mb-6 relative" ref={subjectDropdownRef}>
```
Replace the dropdown with a text box

midhunann commented:
After replacing the subject dropdown into a textbox we have changed from using a array of subjects to a single subject so I have also modified the backend/controllers/classController.js to parse a single string subject instead of a json object which contains an array of subjects.

Aashiq-Edavalapati reviewed (approved):
LGTM👍🏻
